Sming/Components/MultipartParser/multipart-parser.patch

```diff
@@ -78,8 +94,98 @@ index 981dabb..52e0200 100644
    int is_last = 0;
  
    while(i < len) {
+@@ -142,9 +110,29 @@ size_t multipart_parser_execute(multipart_parser* p, const char *buf, size_t len
+           break;
+         }
+         if (c != p->multipart_boundary[p->index]) {
+-          return i;
++          p->state = s_preamble;
++          --i;
++        } else {
++          p->index++;
++        }
++        break;
++
++      case s_preamble:
```

Comment:
That code should be tested. My very quick test resulted in getting stuck here.

Reply:
Strange, it works for me (tested with the `otatool.py upload` from PR#2006, modified to include some preamble data).
Can you describe your test setup?
(The preamble data must end with '\r\n' before the first boundary)